docs/bricks_design.rst

```diff
@@ -0,0 +1,602 @@
+    >>>         self.weight_matrices = []
+    >>>         self.bias_vectors = []
+    >>>         self.activations = activations
+    >>>         for i in range(len(dims) - 1):
+    >>>             self.weight_matrices.append(
+    >>>                 theano.shared(weight_initializers[i]((dims[i], dims[i + 1]))))
+    >>>         for i in range(len(dims) - 1):
+    >>>             self.bias_vectors.append(theano.shared(
+    >>>                 bias_initializers[i](dims[i + 1])))
+    >>> 
+    >>>     @application
+    >>>     def apply(self, inputs):
+    >>>         outputs = inputs
+    >>>         for layer_number, (dim, activation, weight_matrix, bias_vector) in\
+    >>>                 enumerate(zip(dims[1:], self.activations, 
+    >>>                               self.weight_matrices, self.bias_vectors)):
+    >>>             outputs = activation.apply(outputs.dot(weight_matrix) + bias_vector)
+    >>>         return outputs
+    >>>
+    >>>
+    >>> inputs = tensor.matrix('x')
+    >>> inputs.tag.test_value = numpy.random.uniform(size=(10, 754))
+    >>> # The new, stone activations
+    >>> activations = [Tanh(), Tanh(), Softmax()]
+    >>> stone = MultiLayer(dims, activations, weight_initializers, bias_initializers)
+    >>> outputs_multilayer = stone.apply(inputs)   
+    >>> layer0_outputs, = [v for v in get_all_variables([outputs_multilayer]) 
+    >>>                    if hasattr(v.tag, 'stone') and v.tag.stone == stone.activations[0] 
+    >>>                       and hasattr(v.tag, 'role') and v.tag.role == 'output']             
+    >>> theano.printing.debugprint(layer0_outputs)
+    Elemwise{identity} [@A] 'Tanh_output'   
+     |Elemwise{tanh,no_inplace} [@B] ''   
+       |Elemwise{identity} [@C] 'Tanh_input'   
+         |Elemwise{add,no_inplace} [@D] ''   
+           |dot [@E] ''   
+           | |Elemwise{identity} [@F] 'MultiLayer_input'   
+           | | |x [@G]
+           | |<TensorType(float64, matrix)> [@H]
+           |DimShuffle{x,0} [@I] ''   
+             |<TensorType(float64, vector)> [@J]
+
+
+Done: the activations of the first layer neurons can be extracted
+from the annotated computation graph in a simple way! Such detailed
+access to the internals of computation graphs offers a range of
+opportunities. The basic one that we have just used is that the variable
+of interest can be *found* in the graph. But furthermore, the variable
+can be *replaced* using ``theano.clone``.
+
+    >>> rng = theano.tensor.shared_randomstreams.RandomStreams()
+    >>> outputs_multilayer_regularized = theano.clone(
+    >>>     outputs_multilayer, 
+    >>>     replace={layer0_outputs:
+    >>>              layer0_outputs * rng.binomial(layer0_outputs.shape,
+    >>>                                            p=0.5)})
+
+The reader might have noticed that in the example above we applied
+dropout regularization. Many other typical deep learning regularization
```

Comment:
Remember to modify this line if you introduce Dropout in the previous part of the tutorial